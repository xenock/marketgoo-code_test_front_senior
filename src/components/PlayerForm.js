import React, { useState } from 'react'
import { useDispatch } from 'react-redux'
import axios from 'axios'
import { Input, Field, Button } from '@marketgoo/ola'
import { addPlayerAPI } from '@slices/players.js'
import { toast } from 'react-toastify'
import styles from './PlayerForm.module.css'

axios.defaults.baseURL = 'http://localhost:3000'

const PlayerForm = () => {
  const dispatch = useDispatch()
  const initialPlayer = {
    name: null,
    team: null,
    score: null
  }
  const [player, setPlayer] = useState(initialPlayer)
  const [synchronizing, setSynchronizing] = useState(false)
  const [valid, setValid] = useState(true)

  const handleChange = event => {
    setPlayer({ ...player, [event.target.name]: event.target.value })
  }

  const handleSubmit = event => {
    event.preventDefault()

    const { name, team, score } = player
    if (!name || !team || !score) {
      setValid(false)
    } else {
      setSynchronizing('syncing')
      setValid(true)
      event.target.reset()

      dispatch(addPlayerAPI(player))
        .then(response => {
          toast.success(`ðŸ¦„ ${response.payload.name} added successfully!`)
          setPlayer(initialPlayer)
          setSynchronizing(null)
        })
        .catch(_ => {
          toast.error('ðŸ¦„ There was an error...')
          setPlayer(initialPlayer)
          setSynchronizing(null)
        })
    }
  }

  return (
    <form className={styles.form} onSubmit={handleSubmit}>
      <fieldset>
        <legend className='ola-headline'>Add new players</legend>
        <Field id='name' label='Player name' error={!valid && !player.name}>
          <Input
            name='name'
            placeholder='Player name'
            onChange={handleChange}
          />
        </Field>
        <Field id='team' label='Team name' error={!valid && !player.team}>
          <Input name='team' placeholder='Team name' onChange={handleChange} />
        </Field>
        <Field id='score' label='Team score' error={!valid && !player.score}>
          <Input
            name='score'
            type='number'
            placeholder='Team score'
            min='0'
            onChange={handleChange}
          />
        </Field>
        <Button busy={synchronizing} variant='primary'>
          Add
        </Button>
      </fieldset>
    </form>
  )
}

export default PlayerForm
