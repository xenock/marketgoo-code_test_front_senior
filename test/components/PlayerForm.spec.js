import React from 'react'
import renderer from 'react-test-renderer'

import PlayerForm from '@components/PlayerForm.js'

jest.mock('react-redux', () => ({
  useDispatch: () => jest.fn()
}))

describe('PlayerForm snapshots', () => {
  it('renders', () => {
    const tree = renderer.create(<PlayerForm />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
