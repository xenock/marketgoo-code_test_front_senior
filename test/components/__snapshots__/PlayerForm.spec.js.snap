// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerForm snapshots renders 1`] = `
<form
  onSubmit={[Function]}
>
  <fieldset>
    <legend>
      Add new players
    </legend>
    <div
      className="ola_field"
    >
      <label
        className="ola_field-label"
        htmlFor="name"
      >
        Player name
      </label>
      <div
        className="ola_field-input"
      >
        <input
          className="ola_input"
          disabled={false}
          id="name"
          name="name"
          onChange={[Function]}
          placeholder="Player name"
          type="text"
        />
      </div>
    </div>
    <div
      className="ola_field"
    >
      <label
        className="ola_field-label"
        htmlFor="team"
      >
        Team name
      </label>
      <div
        className="ola_field-input"
      >
        <input
          className="ola_input"
          disabled={false}
          id="team"
          name="team"
          onChange={[Function]}
          placeholder="Team name"
          type="text"
        />
      </div>
    </div>
    <div
      className="ola_field"
    >
      <label
        className="ola_field-label"
        htmlFor="score"
      >
        Team score
      </label>
      <div
        className="ola_field-input"
      >
        <input
          className="ola_input"
          disabled={false}
          id="score"
          min="0"
          name="score"
          onChange={[Function]}
          placeholder="Team score"
          type="number"
        />
      </div>
    </div>
    <button
      className="ola_button is-primary"
      disabled={false}
    >
      <span
        className="ola_button-text"
      >
        Add
      </span>
    </button>
  </fieldset>
</form>
`;
